src/lib/alloc.c

```diff
@@ -153,41 +153,36 @@ static void init_heap_map(struct mm_heap *heap, int count)
 }
 
 /* allocate from system memory pool */
-static void *rmalloc_sys(uint32_t flags, int caps, int core, size_t bytes)
+static void *rmalloc_sys(struct mm_heap *heap, uint32_t flags, int caps, size_t bytes)
 {
-	struct mm *memmap = memmap_get();
 	void *ptr;
-	struct mm_heap *cpu_heap;
 	size_t alignment = 0;
 
-	/* use the heap dedicated for the selected core */
-	cpu_heap = memmap->system + core;
-	if ((cpu_heap->caps & caps) != caps)
+	if ((heap->caps & caps) != caps)
```

Comment:
We should probably make these requests survivable and report an error rather than panic now sinec we are reusing this code for other heaps.

Reply:
This would break this assumption:

> 1) System Zone. Fixed size heap where alloc always succeeds and is never
 * freed

Reply:
ok, but isn't the intention to reuse this code block for all heaps/zones now ?

Reply:
it's reused for all system zones (`SOF_MEM_ZONE_SYS_SHARED` and `SOF_MEM_ZONE_SYS`), runtime zones have much different implementation.